consensus/core/src/networktype.rs

```diff
@@ -104,3 +97,124 @@ impl Display for NetworkType {
         f.write_str(s)
     }
 }
+
+#[derive(thiserror::Error, PartialEq, Eq, Debug, Clone)]
+pub enum NetworkIdError {
+    #[error("Invalid network name prefix: {0}. The expected prefix is 'kaspa'.")]
+    InvalidPrefix(String),
+
+    #[error(transparent)]
+    InvalidNetworkType(#[from] NetworkTypeError),
+
+    #[error("Invalid network suffix: {0}. Only 32 bits unsigned integer (u32) are supported.")]
+    InvalidSuffix(String),
+}
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, BorshSerialize, BorshDeserialize, BorshSchema, PartialEq, Eq)]
+#[serde(rename_all = "lowercase")]
+pub struct NetworkId {
+    pub network_type: NetworkType,
+    pub suffix: Option<u32>,
+}
+
+impl NetworkId {
+    pub fn new(network_type: NetworkType, suffix: Option<u32>) -> Self {
```

Comment:
I'd call it `with_prefix` and make `new(network_type: NetworkType)` which defaults to no prefix. Also make both of them const and use them when initiating MAINNET/DEVNET params etc

Reply:
You mean `with_suffix`, right?

Reply:
yes sure